srml/staking/src/lib.rs

```diff
@@ -528,12 +531,14 @@ decl_storage! {
 		/// The current era index.
 		pub CurrentEra get(current_era) config(): EraIndex;
 
-		/// Maximum reward, per validator, that is provided per acceptable session.
-		pub CurrentSessionReward get(current_session_reward) config(): BalanceOf<T>;
+		/// The accumulated reward for the current era. Reset to zero at the beginning of the era and
+		/// increased for every successfully finished session.
+		pub CurrentEraStart get(current_era_start): T::Moment;
 
-		/// The accumulated reward for the current era. Reset to zero at the beginning of the era
-		/// and increased for every successfully finished session.
-		pub CurrentEraReward get(current_era_reward): BalanceOf<T>;
+		/// A vec of validator to reward and their associated number of points.
+		/// A validator can appear in many elements of this vec.
+		// TODO TODO: should we use such a structure ? aggregate with an existing one ? use a map ?
+		pub CurrentEraRewards: Vec<(T::AccountId, u32)>;
```

Comment:
an indexed `Vec<u32>` is also fine as long as this doesn't live beyond eras since we can identify validators by their index into the session's `Vec<AccountId>` validator series.